src/_http/http_client.cpp

```diff
@@ -41,16 +42,19 @@ using Poco::Net::SSLManager;
 
 
 HTTPClient::HTTPClient() {
-    HTTPSessionInstantiator::registerInstantiator();
-    HTTPSSessionInstantiator::registerInstantiator();
-
-    // Prepare for SSLManager
-    Poco::SharedPtr<AcceptCertificateHandler> cert =
-        new AcceptCertificateHandler(false);
-    const Poco::Net::Context::Ptr context = new Context(
-        Context::CLIENT_USE, "", "", "", Context::VERIFY_NONE, 9, false,
-        "ALL:!ADH:!LOW:!EXP:!MD5:@STRENGTH");
-    SSLManager::instance().initializeClient(0, cert, context);
+    try {
+        HTTPSessionInstantiator::registerInstantiator();
+        HTTPSSessionInstantiator::registerInstantiator();
+
+        // Prepare for SSLManager
+        Poco::SharedPtr<AcceptCertificateHandler> cert =
+            new AcceptCertificateHandler(false);
+        const Poco::Net::Context::Ptr context = new Context(
+            Context::CLIENT_USE, "", "", "", Context::VERIFY_NONE, 9, false,
+            "ALL:!ADH:!LOW:!EXP:!MD5:@STRENGTH");
+        SSLManager::instance().initializeClient(0, cert, context);
+    } catch (std::exception& e) {
+    }
```

Comment:
Not quite sure here. Do you expect the object to still be usable if an exception is thrown during that initialisation? 

If not, let it throw to signal to users that they cannot use it. Remember that when constructors fail destructors do not get called, so you may end up with a memory leak (this can be fixed by either using smart pointers or creating a separate `initialise()` method). 

If you still want the object to be used despite initialisation errors, let's at least add a comment saying this is the case. I'd also log the exception somehow, at least `std::cerr << e.what();`.

Reply:
Don't expect an exception. Removed the `try` block.